corehq/apps/locations/dbaccessors.py

```diff
@@ -40,3 +44,25 @@ def get_all_users_by_location(domain, location_id):
         reduce=False,
     )
     return (CouchUser.wrap_correctly(res['doc']) for res in results)
+
+
+def users_have_locations(domain):
+    from corehq.apps.users.models import CouchUser
+    return bool(CouchUser.get_db().view(
+        'users_extra/users_by_location_id',
+        startkey=[domain],
+        endkey=[domain, {}],
+        reduce=True,
+    ).one())
+
+
+def get_users_assigned_to_locations(domain):
+    from corehq.apps.users.models import CouchUser
+    ids = [res['id'] for res in CouchUser.get_db().view(
+        'users_extra/users_by_location_id',
+        startkey=[domain],
+        endkey=[domain, {}],
+        include_docs=False,
+        reduce=False,
```

Comment:
This one does need up-to-the-minute data - it's used to figure out which users need to be unassigned.